### PR TITLE
feat: DBTP-612 allow wildcard in branch names

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**

### CODE REVIEW

File: dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts [hunk 1]
```diff
@@ -54,9 +54,17 @@ export class TransformedStack extends cdk.Stack {
         );
 
         for (const branch of watchedBranches) {
+            let sanitisedBranch = branch;
+
+            if (branch?.endsWith('*')){
+                sanitisedBranch?.replace('*', '.*');
+            } else {
+                sanitisedBranch = sanitisedBranch?.concat('$');
+            }
```
Comment (samuele-mattiuzzo) [inline dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts:63]: @acodeninja I think this is enough, but I also recall our conversation around the branch having to allow for `feat-whatever` and `feat` and I think this is what covers it

File: dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts [hunk 2]
```diff
@@ -54,9 +54,17 @@ export class TransformedStack extends cdk.Stack {
         );
 
         for (const branch of watchedBranches) {
+            let sanitisedBranch = branch;
```
Comment (gabelton) [inline dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts:57]: Is there any way of extracting and unit testing this logic to check it covers the edge cases described on the ticket?

Comment (gabelton) [reply dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts:57]: Or maybe you can get away with just adding a few more scenarios / parameters to your existing `test_command_pipeline` tests?

Comment (samuele-mattiuzzo) [reply dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts:57]: I hope so, it's something I'm looking into now. None of `stack.ts` is tested here, so I'll see and assess, if it's a big task I might add a separate ticket for it

Comment (samuele-mattiuzzo) [reply dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts:57]: @gabelton yes valid suggestion, that would however test the regex not the changes made in `stack.ts`.

I still want to refactor the validation tests a little bit so I still have some work to do on this

**Review by acodeninja — APPROVED** [review]
LGTM

**Pull request merged**